build-system/tasks/check-types.js

```diff
@@ -221,6 +222,44 @@ async function typeCheck(targetName) {
   log(green('SUCCESS:'), 'Type-checking passed for target', cyan(targetName));
 }
 
+/**
+ * Typechecks all JS files within src and extensions for a specific set
+ * of closure errors.
+ *
+ * This is a temporary measure while we restore 100% typechecking.
+ *
+ * @param {Array<string>} entryPoints
+ */
+async function ensureMinTypeQuality(entryPoints) {
+  const srcGlobs = ['src/**/*.js', 'extensions/**/*.js'];
+  const errorsToCheckFor = [
+    'JSC_BAD_JSDOC_ANNOTATION',
+    'JSC_INVALID_PARAM',
+    'JSC_TYPE_PARSE_ERROR',
+  ];
+
+  let msg = '';
+  await closureCompile(entryPoints, './dist', `check-types.js`, {
+    include3pDirectories: false,
+    includePolyfills: false,
+    typeCheckOnly: true,
+    extraGlobs: globby.sync(srcGlobs),
+    warningLevel: 'VERBOSE',
+    logger: (str) => (msg = str),
+  }).catch(() => {});
+
+  const targetErrors = msg
+    .split('\n')
+    .filter((s) => errorsToCheckFor.some((error) => s.includes(error)))
+    .join('\n')
+    .trim();
```

Comment:
Instead of overriding `logger` and scraping the results, I have an alternative to suggest:

- Set three new fields in `options` for `jscomp_error`, `jscomp_warning`, and `jscomp_off`
- Pass them in as part of the call to `closureCompile()` on line 242
- In `generateCompilerOptions()` in `compile.js`, override the hard-coded defaults if passed in

Assuming this will actually serve your purpose, I think it's worth the trouble as it will permanently make type-checking more flexible with the `jscomp_` levels. WDYT?

Reply:
> Set three new fields in options for jscomp_error, jscomp_warning, and jscomp_off

jscom_ levels are actually less fine grained than this. The error groupings within closure aren't too useful imo

Reply:
From my reading of the code, we don't use those error groupings. Instead, we enable / disable individual errors as follows: 

https://github.com/ampproject/amphtml/blob/6447a642edbeead3d01410946e857b5600cdb5bd/build-system/compile/compile.js#L297-L323

I'm suggesting that we also support overriding these lists via `options`. Does this seem reasonable?

Reply:
Right, and if you look at all the [eligible categories](https://github.com/google/closure-compiler/wiki/Warnings#warnings-categories), examples include: `checkVars`, `checkTypes`, `const` etc.

Some are super specific, like `const`: "Warnings when a variable or member property marked @const is reassigned.".
But then theres `checkTypes` with the description: "Type-checking". 

Diving deeper, the closure source calls each of these [DiagnosticGroup](https://github.com/google/closure-compiler/blob/7ff6e25843097791416d1544c88dc8711f6be64a/src/com/google/javascript/jscomp/DiagnosticGroups.java#L317)s. The `typeCheck` group is > 40 individual rules within it.

As-written, the code allows for individually checking the rules _much_ more granularly.

---

All that said, I do agree that it would be an overall good change to have the these three fields configurable. It just wouldn't be helpful for this specific case.

Reply:
Your latest explanation has me convinced. 👍